curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md

```diff
@@ -0,0 +1,253 @@
+```
+
+You should define a function `addLunchToStart`.
+
+```js
+assert.isFunction(addLunchToStart);
+```
+
+The function `addLunchToStart` should take a string as an argument.
+
+```js
+assert.match(addLunchToStart.toString(), /\s*addLunchToStart\(\s*\w+\s*\)/);
+```
+
+`addLunchToStart("Sushi")` should return the string `"Sushi added to the start of the lunch menu."`.
+
+```js
+assert.strictEqual(addLunchToStart("Sushi"), "Sushi added to the start of the lunch menu.");
+
+// prevent hardcoding
+assert.strictEqual(addLunchToStart("Salad"), "Salad added to the start of the lunch menu.");
+assert.strictEqual(addLunchToStart("Pasta"), "Pasta added to the start of the lunch menu.");
+```
+
+You should define a function `removeLastLunch`.
+
+```js
+assert.isFunction(removeLastLunch)
+```
+
+When the `lunches` array is empty, the function `removeLastLunch` should return the string `"No lunches to remove."`.
+
+```js
+lunches = [];
+assert.strictEqual(removeLastLunch(), "No lunches to remove.");
+```
+
+When `lunches = ["Stew", "Soup", "Toast"]`, the function `removeLastLunch` should return the string `"Toast removed from the end of the lunch menu."`.
+
+```js
+lunches = ["Stew", "Soup", "Toast"];
+assert.strictEqual(removeLastLunch(), "Toast removed from the end of the lunch menu.");
+
+// prevent hardcoding
+assert.strictEqual(removeLastLunch(), "Soup removed from the end of the lunch menu.");
+assert.strictEqual(removeLastLunch(), "Stew removed from the end of the lunch menu.");
+```
+
+When the `lunches` array is empty, the function `removeFirstLunch` should return the string `"No lunches to remove."`.
+
+```js
+lunches = [];
+assert.strictEqual(removeFirstLunch(), "No lunches to remove.");
+```
+
+When `lunches = ["Salad", "Eggs", "Cheese"]`, the function `removeFirstLunch` should return the string `"Salad removed from the start of the lunch menu."`.
+
+```js
+lunches = ["Salad", "Eggs", "Cheese"];
+assert.strictEqual(removeFirstLunch(), "Salad removed from the start of the lunch menu.");
+
+// prevent hardcoding
+assert.strictEqual(removeFirstLunch(), "Eggs removed from the start of the lunch menu.");
+assert.strictEqual(removeFirstLunch(), "Cheese removed from the start of the lunch menu.");
+```
+
+You should define a function `getRandomLunch`.
+
+```js
+assert.isFunction(getRandomLunch);
+```
+
+When the `lunches` array is empty, the function `getRandomLunch` should return the string `"No lunches available."`.
+
+```js
+lunches = [];
+assert.strictEqual(getRandomLunch(), "No lunches available.");
+```
+
+When `lunches = ["Fish", "Fries", "Roast"]`, the function `getRandomLunch` should return a string in the format `"Randomly selected lunch: [Lunch Item]"`.
+
+```js
+let initialCondition = "false";
+lunches = ["Fish", "Fries", "Roast"]
+
+lunches.forEach((lunch) => {
+  const result = getRandomLunch();
+  if(result === `Randomly selected lunch: ${lunch}`){
+    initialCondition = "true";
+  }
+});
+assert(initialCondition);
+```
```

Comment:
this test always passes, you have `assert("any string")` which always passes.
You need to convert `initialCondition` to a boolean, but then it can also randomly not pass.

I suggest using the new mockRandom thing so you know which item should come out, and you know which item should come out of `getRandomLunch`. Or you can even force `math.random` to come out with a specific value adding before the call to `getRandomLunch` one of these:
```js
math.random = () => 0; //to get the first item in the array
math.random = () => 0.4; // to get the second item in the array
math.random = () => 0.8; // to get the third item in the array
```

Reply:
```suggestion
const temp = Math.random;

lunches = ["Fish", "Fries", "Roast"];

// check that it correctly outputs the first item
Math.random = () => 0;
assert.equal(getRandomLunch(), `Randomly selected lunch: ${lunches[0]}`);

// second item
Math.random = () => 0.4;
assert.equal(getRandomLunch(), `Randomly selected lunch: ${lunches[1]}`);

// third item
Math.random = () => 0.8;
assert.equal(getRandomLunch(), `Randomly selected lunch: ${lunches[2]}`);

// restore Math.random
Math.random = temp;
```